dipy/direction/closest_peak.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+from dipy.direction.peaks import peak_directions, default_sphere
+from dipy.reconst.shm import order_from_ncoef, sph_harm_lookup
+from dipy.tracking.local.direction_getter import DirectionGetter
+from dipy.tracking.local.interpolation import trilinear_interpolate4d
+
+
+def _closest_peak(peak_directions, prev_step, cos_similarity):
+    """Return the closest direction to prev_step from peak_directions.
+
+    All directions should be unit vectors. Antipodal symmetry is assumed, ie
+    direction x is the same as -x.
+
+    Parameters
+    ----------
+    peak_directions : array (N, 3)
+        N unit vectors.
+    prev_step : array (3,) or None
+        Previous direction.
+    cos_similarity : float
+        `cos(max_angle)` where `max_angle` is the maximum allowed angle between
+        prev_step and the returned direction.
+
+    Returns
+    -------
+    direction : array or None
+        If prev_step is None, returns peak_directions. Otherwise returns the
+        closest direction to prev_step. If no directions are close enough to
+        prev_step, returns None
+    """
+    peak_dots = np.dot(peak_directions, prev_step)
+    closest_peak = abs(peak_dots).argmax()
+    dot_closest_peak = peak_dots[closest_peak]
+    if dot_closest_peak >= cos_similarity:
+        return peak_directions[closest_peak]
+    elif dot_closest_peak <= -cos_similarity:
+        return -peak_directions[closest_peak]
+    else:
+        return None
+
+
+class PmfGen(object):
+    pass
+
+
+class SimplePmfGen(PmfGen):
+
+    def __init__(self, pmf_array):
+        if pmf_array.min() < 0:
+            raise ValueError("pmf should not have negative values")
+        self.pmf_array = pmf_array
+
+    def get_pmf(self, point):
+        return trilinear_interpolate4d(self.pmf_array, point)
+
+
+class SHCoeffPmfGen(PmfGen):
+
+    def __init__(self, shcoeff, sphere, basis_type):
+        self.shcoeff = shcoeff
+        self.sphere = sphere
+        sh_order = order_from_ncoef(shcoeff.shape[-1])
+        try:
+            basis = sph_harm_lookup[basis_type]
+        except KeyError:
+            raise ValueError("%s is not a known basis type." % basis_type)
+        self._B, m, n = basis(sh_order, sphere.theta, sphere.phi)
+
+    def get_pmf(self, point):
+        coeff = trilinear_interpolate4d(self.shcoeff, point)
+        pmf = np.dot(self._B, coeff)
+        return pmf
+
+
+class BaseDirectionGetter(DirectionGetter):
+    """A base class for dynamic direction getters"""
+
+    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=.1, **kwargs):
```

Comment:
call `__init__` of super class is missed. you should add something like : 

`super(BaseDirectionGetter, self).__init__(pmf_gen, max_angle, sphere, pmf_threshold, **kwargs)`

Reply:
Calling the super of the parent class is not required in python and we generally only do it if there is a reason.